Merge "Getting Started with Liberty Project blog" into staging (#3770)
OpenLiberty/blogs: posts/2024-05-05-liberty-project-starter-guide-IntelliJ.adoc

title: "Developing a cloud-native Java application with the Liberty Starter, Liberty Tools, and IntelliJ IDEA"
# Do NOT change the categories section
categories: blog
author_picture: https://avatars3.githubusercontent.com/Ruilin-Ma
author_github: https://github.com/Ruilin-Ma
seo-title: Developing a cloud-native Java application with the Liberty Starter, Liberty Tools, and IntelliJ IDEA - OpenLiberty.io
seo-description: Learn how to create a Liberty Maven project for a new cloud-native Java application by using the Open Liberty Starter, and to edit your application using the Liberty Tools in IntelliJ IDEA.
blog_description: Learn how to create a Liberty Maven project for a new cloud-native Java application by using the Open Liberty Starter, and to edit your application using the Liberty Tools in IntelliJ IDEA.
open-graph-image: https://openliberty.io/img/blog/liberty-starter.png
open-graph-image-alt: Open Liberty starter
---
= Developing a cloud-native Java application with Liberty Tools in IntelliJ IDEA
Ruilin Ma <https://github.com/Ruilin-Ma>
:imagesdir: /
:url-prefix:
:url-about: /

:figure-caption!:
//Blank line here is necessary before starting the body of the post.

> Given the rapid rate of change in technology, the array of tools now available, and the rising complexity of cloud-native applications, we understand that it can often be difficult to know how or even where to get started creating a cloud-native Java application. To make this easier, we've put together this short blog to lead you through how you can easily create a simple cloud-native Java application and edit it effectively within an IDE, all within just 10 minutes! Read on to learn how to create a Liberty Maven project for a new cloud-native Java application by using the Open Liberty Starter and to edit your application using the Liberty Tools in IntelliJ IDEA.

Reviewer: ```suggestion
Given the rapid rate of technological change, the array of tools available, and the rising complexity of cloud-native applications, it can be difficult to know how to start creating a cloud-native Java application. This short post to leads you through creating a simple cloud-native Java application and editing it within an IDE, all within just 10 minutes! Read on to learn how to easily create a Liberty Maven project for a new cloud-native Java application by using the Open Liberty Starter and to edit your application using the Liberty Tools in IntelliJ IDEA.
```

Liberty Tools offer three ways to start your Liberty application in dev mode: Start, Start with configuration, or Start in a container.

For now, click the **Start** action to simply start your application in dev mode.

Dev mode automatically detects, recompiles, and deploys code changes whenever you save a new change in your IDE or text editor. The following example creates a simple REST resource Java file.

Ensure that Liberty dev mode is running, then create the following Java class file named `HelloWorldResource.java` as the REST resource:

[source, java]
----
src/main/java/com/demo/rest/HelloWorldResource.java
----

Paste the following code into the file: 

[source,java]
```
package com.demo.rest;

import jakarta.ws.rs.GET;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.Produces;
import jakarta.ws.rs.core.MediaType;

@Path("/hello")
public class HelloWorldResource {

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String helloWorld() {
        return "Hello, World!";
    }
}
```

> When the console displays `Web application available`, the Liberty server has successfully detected, recompiled, and deployed the changes. You can now view the message drafted in the example by accessing the link: http://localhost:9080/app-name/api/hello

Reviewer: ```suggestion
When the console displays `Web application available`, the Liberty server has successfully detected, recompiled, and deployed the changes. You can now view the message drafted in the example by accessing the following link: http://localhost:9080/app-name/api/hello
```

For more information, see:

* link:https://openliberty.io/docs/latest/development-mode.html[Open Liberty dev mode docs]
* link:https://developer.ibm.com/articles/awb-effective-cloud-native-development-open-liberty-intellij-idea/[Effective cloud-native Java app development with Open Liberty in IntelliJ IDEA]
* link:https://github.com/OpenLiberty/liberty-tools-intellij/blob/main/docs/user-guide.md#run-your-application-on-liberty-using-dev-mode[Liberty Tools for IntelliJ IDEA user guide]

== Next steps

> In just 10 minutes, if you have been following along with the instructions in this blog, you have now successfully been able to create a cloud-native Java application, deploy it, and edit it in an agile manner with rapid feedback. Take your learning to the next level and further develop your application by exploring more of Open Liberty's features or exploring the APIs supported by open source, industry standard projects like MicroProfile and Jakarta EE through our link:https://openliberty.io/guides/[interactive guides].

Reviewer: ```suggestion
In just 10 minutes, if you followed along with the instructions in this post, you successfully created a cloud-native Java application, deployed it, and edited it in an agile manner, with rapid feedback. Take your learning to the next level and further develop your application by exploring more of Open Liberty's features or exploring the APIs supported by open source, industry standard projects like MicroProfile and Jakarta EE through our link:https://openliberty.io/guides/[interactive guides].
```

blog_description: Learn how to create a Liberty Maven project for a new cloud-native Java application by using the Open Liberty Starter, and to edit your application using the Liberty Tools in IntelliJ IDEA.
open-graph-image: https://openliberty.io/img/blog/liberty-starter.png
open-graph-image-alt: Open Liberty starter
---
= Developing a cloud-native Java application with Liberty Tools in IntelliJ IDEA
Ruilin Ma <https://github.com/Ruilin-Ma>
:imagesdir: /
:url-prefix:
:url-about: /

:figure-caption!:
//Blank line here is necessary before starting the body of the post.

Learn how to create a Liberty Maven project for a new cloud-native Java application by using the Open Liberty Starter, and to edit your application using the Liberty Tools in IntelliJ IDEA.

* <<prerequisites, Before you start>>
* <<libertyStarter, Create a Liberty Maven project with the Open Liberty Starter>>
* <<ImportProject, Import your project into IntelliJ IDEA>>
* <<AboutProject, Explore your Liberty Maven project>>
* <<libertyToolsWithDevMode, Edit your app using Liberty dev mode>>

[#prerequisites]
== Before you start

=== Installing Java

> Install either Java 17 or Java 21. You can link:https://www.ibm.com/support/pages/semeru-runtimes-installation[download IBM Semeru Runtimes for free]. Semeru integrates well with Liberty, but you can find out more about Semeru and other Java distributionslink:https://foojay.io/today/where-do-you-get-your-java/[in this article].

Reviewer: ```suggestion
Install either Java 17 or Java 21. You can link:https://www.ibm.com/support/pages/semeru-runtimes-installation[download IBM Semeru Runtimes for free]. Semeru integrates well with Liberty, but you can find out more about Semeru and other Java distributions link:https://foojay.io/today/where-do-you-get-your-java/[in this article].
```

Author: Hi @dmuelle, sorry I missed your message. This issue has resolved in the [commit](https://github.com/OpenLiberty/blogs/pull/3770/commits/904685cfff1956e63ded530362b4dbb01a42006f), Thank you
